maven/poms.bzl

```diff
@@ -45,38 +65,223 @@ def _process_dependency(dep_node):
-            dependencies = node.children
-    return [_process_dependency(x) for x in dependencies]
+        if node.label == labels.PROPERTIES:
+            properties_nodes = node.children
+    properties = {}
+    for node in properties_nodes:
+        properties[node.label] = node.content
+    return properties
 
 def _format_dependency(dep):
     result = "%s:%s:%s" % (dep.group_id, dep.artifact_id, dep.version)
     if bool(dep.classifier):
         type = dep.type if bool(dep.type) else "jar"
         result = "%s:%s" % (result, dep.type)
-    else:
-        if bool(dep.type) and not dep.type == "jar":
-            result = "%s:%s" % (result, dep.type)
+    elif bool(dep.type) and not dep.type == "jar":
+        result = "%s:%s" % (result, dep.type)
     return result
 
 def _parse(xml_text):
     root = xml.parse(xml_text)
     for node in root.children:
-        if node.label == "project":
+        if node.label == labels.PROJECT:
             return node
     fail("No <project> tag found in supplied xml: %s" % xml)
 
+# A pom-specific node-merge algorith,
+def _merge_content_last_wins(a, b):
+    if not bool(a):
+        return b
+    elif not bool(b):
+        return a
+    else:
+        if a.label != b.label:
+            fail("Attempt to merge to different pom elements: %s, %s", (a, b))
+        return xml.new_node(
+            label = a.label,
+            content = b.content if bool(b.content) else a.content,
+        )
+
+# This could be 100% reusable, except for the limit on recursion.  The strategy can't loop back and call this. :/
+def _merge_leaf_elements(parent_list, child_list):
+    index = {}
+    merged = []
+    for i in range(len(parent_list)):
+        merged.append(parent_list[i])
+        index[parent_list[i].label] = i
+    for i in range(len(child_list)):
+        index_of_property = index.get(child_list[i].label, -1)
+        if index_of_property >= 0:
+            merged[index_of_property] = child_list[i]
+        else:
+            merged.append(child_list[i])
+            index[child_list[i].label] = len(merged)
+    return merged
+
+def _children_if_exists(node):
+    return node.children if bool(node) else []
+
+def _merge_properties_section(parent_node, child_node):
+    if not bool(parent_node):
+        return child_node
+    elif not bool(child_node):
+        return parent_node
+    children = _merge_leaf_elements(
+        _children_if_exists(xml.find_first(parent_node, labels.PROPERTIES)),
+        _children_if_exists(xml.find_first(child_node, labels.PROPERTIES)),
+    )
+    return xml.new_node(label = labels.PROPERTIES, children = children)
+
+# Description:
+#   Merges the dependency section of the pom.  This makes an assumption that deps sections won't have both the main
+#   artifact for a group_id/artifact_id pair AND one of the same pom's classified artifacts.  It is possible, and in
+#   those cases, the deps will be wrong and the build snippet should be explicitly substituted.
```

Comment:
How will users know that this is the situation and what to do about it?

Reply:
Presently, they'll just fail to have the right deps.  I did some sampling and I couldn't find an example of this in the corpus. I'm filing a bug to investigate and either make this possible, or to detect and report.

Reply:
Filed as #32